### ISSUE TITLE
clang doesn't apply -Wconversion flag when including header from include search path

### ISSUE THREAD

**Issue by inicula** [llvm/llvm-project#82645]
* Scenario no. 1 (everything is as expected, i.e. the narrowing conversion is caught at compile-time):

```c
/* file.c */

/* Compile with: clang -std=c11 file.c -Wall -Wextra -Wpedantic -Wconversion -Werror */

#include <assert.h>
#include <inttypes.h>
#include <unistd.h>

#define SOME_WEIRD_MACRO(type, a, b)                                           \
  do {                                                                         \
    type tmp1 = a;                                                             \
    type tmp2 = b;                                                             \
    return tmp1 == tmp2;                                                       \
  } while (0)

int main(void) {
  unsigned long long size = (unsigned long long)-1;
  SOME_WEIRD_MACRO(int, 0, size); // ERROR (see below)
}

/*

file.c:16:28: error: implicit conversion loses integer precision: 'unsigned long long' to 'int' [-Werror,-Wshorten-64-to-32]
  SOME_WEIRD_MACRO(int, 0, size);
  ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
file.c:10:17: note: expanded from macro 'SOME_WEIRD_MACRO'
    type tmp2 = b;                                                             \
         ~~~~   ^
1 error generated.

*/

```

* Scenario no. 2 (presumed bug: the narrowing conversion is not caught at compile-time; no errors, no warnings):

```c
/* /usr/local/include/my_weird_macro.h */

#include <assert.h>
#include <inttypes.h>
#include <unistd.h>

#define SOME_WEIRD_MACRO(type, a, b)                                           \
  do {                                                                         \
    type tmp1 = a;                                                             \
    type tmp2 = b;                                                             \
    return tmp1 == tmp2;                                                       \
  } while (0)

```

```c
/* file.c */

/* Compile with: clang -std=c11 file.c -Wall -Wextra -Wpedantic -Wconversion -Werror */

#include <my_weird_macro.h>

int main(void) {
  unsigned long long size = (unsigned long long)-1;
  SOME_WEIRD_MACRO(int, 0, size); // OK for some reason
}

```

Note: if scenario no. 2 is modified such that `my_weird_macro.h` is included from the same directory as `file.c` (i.e. moving it locally and doing `#include "my_weird_macro.h"` instead of `#include <my_weird_macro.h>`), then the compiler reports the conversion error.

Version info (package downloaded from: https://github.com/llvm/llvm-project/releases/download/llvmorg-17.0.6/clang+llvm-17.0.6-x86_64-linux-gnu-ubuntu-22.04.tar.xz):

```
clang version 17.0.6 (https://github.com/llvm/llvm-project 6009708b4367171ccdbf4b5905cb6a803753fe18)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/bin/clang-17/bin
```

**Comment by shafik** [comment]
I believe this is b/c it may be considering it a system header: https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-in-system-headers

and we suppress diagnostics in system headers but you have options for working around that as the link above describes.
